### PR TITLE
21 08 30

### PR DESCRIPTION
**Pull request by lindajohnstone**

### CODE REVIEW

File: MarsRover.Tests/ControllerShould.cs [hunk 1]
```diff
@@ -25,48 +25,17 @@ public void Setup_ReturnsRoverLocationAndDirection_GivenValidMapAndRoverData()
                 .Returns("2,0");
             var output = new StubOutput();
             var mockMapInput = new Mock<IMapInput>();
-            mockMapInput.Setup(_ => _.Read("TestFiles/validfile1.txt"));
-            var map = SetMap();
-            var rover = new Rover(Direction.North, 2, 0);
+            mockMapInput.Setup(_ => _.Read("TestFiles/validFile1.txt"))
```
Comment (indianbread) [inline MarsRover.Tests/ControllerShould.cs:28]: i recommend either using the real map input or making your own stub to make use of the contents of your text files

File: MarsRover.Tests/ControllerShould.cs [hunk 2]
```diff
@@ -25,48 +25,17 @@ public void Setup_ReturnsRoverLocationAndDirection_GivenValidMapAndRoverData()
...
-                new Square(SquareContent.None, 4, 2),
-                new Square(SquareContent.None, 0, 3),
-                new Square(SquareContent.None, 1, 3),
-                new Square(SquareContent.Obstacle, 2, 3),
-                new Square(SquareContent.None, 3, 3),
-                new Square(SquareContent.None, 4, 3)
-            };
-            var map = new Map(width, height, squares);
-            return map;
+            controller.Rover.Direction.Should().Be(Direction.North);
```
Comment (indianbread) [inline MarsRover.Tests/ControllerShould.cs:36]: is there much value in testing these things in the controller? The parsing is already covered in your parser tests and i remember seeing something similar already tested in your rover test class. There's some 'reaching in' going on here.

File: MarsRover.Tests/GeneratorShould.cs [hunk 3]
```diff
@@ -11,26 +11,22 @@ public class GeneratorShould
...
+            var rover = new Rover(Direction.North, 2, 0); 
             var generator = new Generator(mockInput.Object, output, mockMapInput.Object, map, rover);
             var expectedDirection = Direction.North;
 
             generator.GenerateRover();
 
             rover.Direction.Should().Be(expectedDirection);
-            rover.Location.X.Should().Be(x);
-            rover.Location.Y.Should().Be(y);
+            rover.Location.X.Should().Be(2);
```
Comment (indianbread) [inline MarsRover.Tests/GeneratorShould.cs:28]: pretty much the same tests in the controller which has already been tested in the rover

File: MarsRover/Controller.cs [hunk 4]
```diff
@@ -58,11 +57,13 @@ validate location
...
-            InitialiseMap(); 
-            InitialiseRover(); // replace with _generator.GenerateRover();
+            Map = InitialiseMap();
+            _output.WriteLine(OutputFormatter.DisplayMap(Map));
+            Rover = InitialiseRover();
+            _output.WriteLine(OutputFormatter.DisplayMap(Map, Rover));
         }
 
-        private void InitialiseMap()
+        private Map InitialiseMap()
```
Comment (kedoodle) [inline MarsRover/Controller.cs:66]: Why change the return type? It's still setting the `Map` prop

File: MarsRover/Controller.cs [hunk 5]
```diff
@@ -58,11 +57,13 @@ validate location
         public void Setup()
         {
             _output.WriteLine(Messages.Title);
-            InitialiseMap(); 
-            InitialiseRover(); // replace with _generator.GenerateRover();
+            Map = InitialiseMap();
```
Comment (kedoodle) [inline MarsRover/Controller.cs:60]: The `Map` prop is already set as part of the `InitialiseMap` function

File: MarsRover/Controller.cs [hunk 6]
```diff
@@ -58,11 +57,13 @@ validate location
         public void Setup()
         {
             _output.WriteLine(Messages.Title);
-            InitialiseMap(); 
-            InitialiseRover(); // replace with _generator.GenerateRover();
+            Map = InitialiseMap();
+            _output.WriteLine(OutputFormatter.DisplayMap(Map));
+            Rover = InitialiseRover();
```
Comment (kedoodle) [inline MarsRover/Controller.cs:62]: Same here with the `InitialiseRover` function

File: MarsRover/MapDisplay.cs [hunk 7]
```diff
@@ -2,7 +2,8 @@ namespace MarsRover
 {
     public class MapDisplay
     {
-        public const string Rover = "🔴";
-        public const string Obstacle = "🪨";
+        public const string Rover = "👹";
```
Comment (kedoodle) [inline MarsRover/MapDisplay.cs:5]: What does the `Rover` look like when it's facing `Direction.North` vs `Direction.South`? Or is it the same?

**Pull request merged**